### PR TITLE
[ANCHOR-339] Add the /event endpoint to Callback API.yml

### PR DESCRIPTION
**Pull request by lijamie98**
This is for the implemenation of [ANCHOR-339](https://stellarorg.atlassian.net/browse/ANCHOR-339)

[ANCHOR-339]: https://stellarorg.atlassian.net/browse/ANCHOR-339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: openapi/anchor-platform/Callbacks API.yml [hunk 1]
```diff
@@ -466,3 +466,34 @@ paths:
             application/json:
               schema:
                 $ref: './schemas.yml#/components/schemas/Error'
+  /events:
```
Comment (JakeUrban) [inline openapi/anchor-platform/Callbacks API.yml:469]: I think we have singular names for the rest of our endpoints in this API right? I agree plural is the standard but we might want to stay consistent.

File: openapi/anchor-platform/Callbacks API.yml [hunk 2]
```diff
@@ -466,3 +466,34 @@ paths:
...
+          application/json:
+            schema:
+              type: object
+              properties:
+                timestamp:
+                    type: string
+                    description: |
+                      `timestamp` the current Unix timestamp (number of seconds since epoch) at the time the callback is sent.
+                payload:
+                  $ref: './schemas.yml#/components/schemas/Event'
```
Comment (JakeUrban) [inline openapi/anchor-platform/Callbacks API.yml:490]: Is now a good time to revisit the schema?

- Is the `id` attribute specific to the event or transaction?
- Why do we have `sep` as a top level attribute when it is available inside the `transaction` object?
- Why do we have a `transaction_error` type? Can't we merge this with `transaction_status_changed`?

Comment (JakeUrban) [reply openapi/anchor-platform/Callbacks API.yml:490]: Also, do we need a timestamp on each event to ensure business can check if they've already processed an event after the one received? Is that possible?

File: openapi/anchor-platform/Callbacks API.yml [hunk 3]
```diff
@@ -466,3 +466,34 @@ paths:
...
+      responses:
+        '200':
+          description: The event is successfully processed and ready to receive next event. The response body is empty.
+        '400':
+          description: The event is invalid or rejected. The response body contains the error message.
```
Comment (JakeUrban) [inline openapi/anchor-platform/Callbacks API.yml:495]: I'm not sure we want to define any non-200 response code. All responses should have 200, and all other response codes will be interpreted as an error and retries will be made according to the AP's strategy.

File: openapi/anchor-platform/Callbacks API.yml (hunk 1 above)
Comment (lijamie98) [reply openapi/anchor-platform/Callbacks API.yml:469]: Changed to singular.

File: openapi/anchor-platform/Callbacks API.yml (hunk 2 above)
Comment (lijamie98) [reply openapi/anchor-platform/Callbacks API.yml:490]: The `Event` schema can be further improved. I think we have a couple of days to review. 

For `id`, `sep`, and the transaction status, I am open to redesign if that's also ok with you.

And we should have a `created_at` in the `Event` schema.

File: openapi/anchor-platform/Callbacks API.yml (hunk 3 above)
Comment (lijamie98) [reply openapi/anchor-platform/Callbacks API.yml:495]: The response code is limited `400`. If the business server returns anything other than 400, they will be considered the same way as `400`. 

The retries can be handled in the next fix version.

File: openapi/anchor-platform/Callbacks API.yml (hunk 2 above)
Comment (JakeUrban) [reply openapi/anchor-platform/Callbacks API.yml:490]: Sounds good to me.

File: openapi/anchor-platform/Callbacks API.yml (hunk 3 above)
Comment (JakeUrban) [reply openapi/anchor-platform/Callbacks API.yml:495]: But 400 itself is not a valid status, so I don't think we should define it in the spec.

**Review by JakeUrban — APPROVED** [review]
Approved with the small comment about 400's

File: openapi/anchor-platform/Callbacks API.yml [hunk 4]
```diff
@@ -466,3 +466,34 @@ paths:
...
+      operationId: postEvent
+      description: |
+        Receive a JSON object representing an event.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                timestamp:
```
Comment (Ifropc) [inline openapi/anchor-platform/Callbacks API.yml:485]: Why is timestamp not a part of an event?

File: openapi/anchor-platform/Callbacks API.yml [hunk 5]
```diff
@@ -466,3 +466,34 @@ paths:
             application/json:
               schema:
                 $ref: './schemas.yml#/components/schemas/Error'
+  /event:
+    post:
+      tags:
+        - Events
+        - SEP-24
+        - SEP-31
+      summary: Receive a event from the anchor platform.
```
Comment (Ifropc) [inline openapi/anchor-platform/Callbacks API.yml:475]: [an event]

File: openapi/anchor-platform/Callbacks API.yml (hunk 2 above)
Comment (Ifropc) [reply openapi/anchor-platform/Callbacks API.yml:490]: If we redesigning event schema in the separate PR this change LGTM

**Review by Ifropc — APPROVED** [review]

**Comment by JakeUrban** [comment]
closing this because these changes have been pulled into https://github.com/stellar/stellar-docs/pull/232

**Pull request closed without merging**